apps/@sparrow-web/src/containers/api/api.common.ts

```diff
@@ -457,74 +457,85 @@ const makeHttpRequestV2 = async (
+  console.table({ url, method, headers, body, contentType });
   const startTime = performance.now();
-  return Promise.race([
-    timeout(apiTimeOut),
-    // Invoke communication
-    invoke("make_http_request_v2", {
-      url,
-      method,
-      headers,
-      body,
-      request,
-    }),
-  ])
-    .then(async (data: string) => {
-      const endTime = performance.now();
-      const duration = endTime - startTime;
-      try {
-        const responseBody = JSON.parse(data);
-        const apiResponse = JSON.parse(responseBody.body);
-        console.table(apiResponse);
-        appInsights.trackDependencyData({
-          id: uuidv4(),
-          name: "RPC Duration Metric",
-          duration,
-          success: true,
-          responseCode: parseInt(apiResponse.status),
-          properties: {
-            source: "frontend",
-            type: "RPC_HTTP",
-          },
-        });
-        return success(apiResponse);
-      } catch (e) {
-        console.error(e);
-        appInsights.trackDependencyData({
-          id: uuidv4(),
-          name: "RPC Duration Metric",
-          duration,
-          success: false,
-          responseCode: 400,
-          properties: {
-            source: "frontend",
-            type: "RPC_HTTP",
-          },
-        });
-        return error("error");
-      }
-    })
-    .catch((e) => {
-      const endTime = performance.now();
-      const duration = endTime - startTime;
-      appInsights.trackDependencyData({
-        id: uuidv4(),
-        name: "RPC Duration Metric",
-        duration,
-        success: false,
-        responseCode: 400,
-        properties: {
-          source: "frontend",
-          type: "RPC_HTTP",
+
+  // Fetch the agent state from local storage
+  const selectedAgent = localStorage.getItem("selectedAgent");
+
+  try {
+    let data;
+    if (selectedAgent === "Cloud Agent") {
+      data = await axios({
+        data: {
+          url,
+          method,
+          headers,
+          body,
+          contentType,
         },
+        url: constants.PROXY_SERVICE,
+        method: "POST",
       });
+    } else {
+      data = await axios({
+        method,
+        url,
+        data: {},
+      });
+    }
+
+    // Abort signal check
+    if (signal?.aborted) {
+      throw new Error(); // Ignore response if request was cancelled
+    }
+
+    const endTime = performance.now();
+    const duration = endTime - startTime;
+
+    try {
+      // Handle response
+      let responseData;
+      if (typeof data.data.data !== "string") {
+        responseData = JSON.stringify(data.data.data);
+      } else {
+        responseData = data.data.data; // Don't modify non-JSON data
+      }
+
+      if (!data.data.status) {
+        throw new Error("Error parsing response");
+      }
+
+      return success({
+        body: responseData,
+        status: data.data.status,
+        headers: data.data.headers,
+      });
+    } catch (e) {
```

Comment:
It can be done without try catch, but I got some edge cases during self testing where there were some probabiltiy of getting errors so to handle it I've kept it deliberately.